src/modeler/AutoRest.Swagger/Validation/XmsPathsInPath.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+using System.Collections.Generic;
+
+namespace AutoRest.Swagger.Validation
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Xms")]
+    public class XmsPathsInPath : TypedRule<ServiceDefinition>
+    {
+        public override IEnumerable<ValidationMessage> GetValidationMessages(ServiceDefinition entity)
+        {
+            if (entity != null && entity.CustomPaths != null)
+            {
+                foreach (var customPath in entity.CustomPaths.Keys)
+                {
+                    var basePath = GetBasePath(customPath);
+                    if (!entity.Paths.ContainsKey(basePath))
+                    {
+                        yield return CreateException(Exception, customPath);
+                    }
+                }
+            }
+
+            yield break;
+        }
```

Comment:
**Linq optimization opportunity**: using `foreach` and `yield return` 
can easily be transformed into a single expression.

from:

``` csharp 
        public override IEnumerable<ValidationMessage> GetValidationMessages(ServiceDefinition entity)
        {
            if (entity != null && entity.CustomPaths != null)
            {
                foreach (var customPath in entity.CustomPaths.Keys)
                {
                    var basePath = GetBasePath(customPath);
                    if (!entity.Paths.ContainsKey(basePath))
                    {
                        yield return CreateException(Exception, customPath);
                    }
                }
            }

            yield break;
        }
```
     
to:
        
``` csharp        
        public override IEnumerable<ValidationMessage> GetValidationMessages(ServiceDefinition entity)
        {
          if (entity != null && entity.CustomPaths != null) {
            return entity.CustomPaths.Keys
              .Where( customPath => !entity.Paths.ContainsKey(basePath) )
              .Select( customPath => CreateException(Exception, customPath) );            
          }
          
          return Enumerable.Empty<ValidationMessage>();          
        }
```

or get really aggressive, and mix in a bit of null-propagation and null-coalescing :
        
``` csharp        
        public override IEnumerable<ValidationMessage> GetValidationMessages(ServiceDefinition entity)
        {
            return entity.?CustomPaths.?Keys
              .Where( customPath => !entity.Paths.ContainsKey(basePath) )
              .Select( customPath => CreateException(Exception, customPath) )
          
              ?? Enumerable.Empty<ValidationMessage>();          
        }
```

The nice thing about the last one is that it really distills out what you're checking for. 

`return` (stuff) `where` (condition) `select` (results)
     
And the null-propagation & coalescing makes it consumer-safe.